app/services/create_github_repo_service.rb

```diff
@@ -248,6 +248,16 @@ def report_template_error_to_failbot(error, options)
     )
   end
 
+  def github_organization_with_access
+    github_client = assignment.creator.github_client
+    starter_code_repository = github_client.repository(assignment.starter_code_repo_id)
+
+    return organization.github_organization if !starter_code_repository.private ||
+        starter_code_repository.owner.login == exercise.organization_login
+
+    GitHubOrganization.new(github_client, starter_code_repository)
+  end
```

Comment:
Awesome! 

Can we use a `attr_accessor`  like `@organization_client` for this? Use the logic to initialize the correct `GitHubOrganization` object, and use that instead of a method call? It can also help us out in testing where we can mock/stub the object.

Another idea would be:

We currently delegate `organization` to `Exercise`, so maybe we can move the logic there? Something on the lines of `Exercise#find_github_organization_with_access(starter_code_repository)` which will modify the `@github_organization` `attr_accessor` on `Exercise` and then we can use it in this service everywhere.

Reply:
Good idea! I combined both your ideas into an instance variable `@github_organization` on `Exercise`